analyzers/src/SonarAnalyzer.CFG/LiveVariableAnalysis/LiveVariableAnalysisBase.cs

```diff
@@ -20,95 +20,94 @@
-                captured.UnionWith(state.Captured);
-                states.Add(block, state);
-                blockLiveIn.Add(block, new HashSet<ISymbol>());
-                blockLiveOut.Add(block, new HashSet<ISymbol>());
-                queue.Enqueue(block);
+                liveOut.UnionWith(successorLiveIn);
             }
-            while (queue.Any())
+            // liveIn = UsedBeforeAssigned + (LiveOut - Assigned)
+            var liveIn = states[block].UsedBeforeAssigned.Concat(liveOut.Except(states[block].Assigned)).ToHashSet();
+            // Don't enqueue predecessors if nothing changed.
+            if (!liveIn.SetEquals(blockLiveIn[block]))
             {
-                Cancel.ThrowIfCancellationRequested();
-
-                var block = queue.Dequeue();
-                var liveOut = blockLiveOut[block];
-                // note that on the PHP LVA impl, the `liveOut` gets cleared before being updated
-                foreach (var successorLiveIn in Successors(block).Select(x => blockLiveIn[x]).Where(x => x.Any()))
+                blockLiveIn[block] = liveIn;
+                foreach (var predecessor in Predecessors(block))
                 {
-                    liveOut.UnionWith(successorLiveIn);
-                }
-                // liveIn = UsedBeforeAssigned + (LiveOut - Assigned)
-                var liveIn = states[block].UsedBeforeAssigned.Concat(liveOut.Except(states[block].Assigned)).ToHashSet();
-                // Don't enqueue predecessors if nothing changed.
-                if (!liveIn.SetEquals(blockLiveIn[block]))
-                {
-                    blockLiveIn[block] = liveIn;
-                    foreach (var predecessor in Predecessors(block))
-                    {
-                        queue.Enqueue(predecessor);
-                    }
+                    queue.Enqueue(predecessor);
                 }
             }
-            if (blockLiveOut[ExitBlock].Any())
-            {
-                throw new InvalidOperationException("Out of exit block should be empty");
-            }
         }
-
-        protected class State
+        if (blockLiveOut[ExitBlock].Any())
         {
-            public ISet<ISymbol> Assigned { get; } = new HashSet<ISymbol>();            // Kill: The set of variables that are assigned a value.
-            public ISet<ISymbol> UsedBeforeAssigned { get; } = new HashSet<ISymbol>();  // Gen:  The set of variables that are used before any assignment.
-            public ISet<ISymbol> ProcessedLocalFunctions { get; } = new HashSet<ISymbol>();
-            public ISet<ISymbol> Captured { get; } = new HashSet<ISymbol>();
-
-            protected State() { }
+            throw new InvalidOperationException("Out of exit block should be empty");
         }
     }
+
+    protected class State
+    {
+        public ISet<ISymbol> Assigned { get; } = new HashSet<ISymbol>();            // Kill: The set of variables that are assigned a value.
```

Comment:
Make them all `HashSet` and use `new()` to initialize

Reply:
I used "[ ]".
Simple and beautiful.